migration-engine/core/src/commands/schema_push.rs

```diff
@@ -0,0 +1,78 @@
+use super::MigrationCommand;
+use crate::parse_datamodel;
+use migration_connector::{DatabaseMigrationMarker, MigrationConnector};
+use serde::{Deserialize, Serialize};
+
+pub struct SchemaPushCommand<'a> {
+    pub input: &'a SchemaPushInput,
+}
+
+#[async_trait::async_trait]
+impl<'a> MigrationCommand for SchemaPushCommand<'a> {
+    type Input = SchemaPushInput;
+    type Output = SchemaPushOutput;
+
+    async fn execute<C, D>(
+        input: &Self::Input,
+        engine: &crate::migration_engine::MigrationEngine<C, D>,
+    ) -> super::CommandResult<Self::Output>
+    where
+        C: MigrationConnector<DatabaseMigration = D>,
+        D: DatabaseMigrationMarker + Send + Sync + 'static,
+    {
+        let connector = engine.connector();
+        let schema = parse_datamodel(&input.schema)?;
+        let inferrer = connector.database_migration_inferrer();
+        let applier = connector.database_migration_step_applier();
+        let checker = connector.destructive_changes_checker();
+
+        let sql_migration = inferrer.infer(&schema, &schema, &[]).await?;
```

Comment:
This is relying on the behaviour of the sql database migration inferrer: introspect, calculate db schema from prisma schema, and work with that (ignoring the steps array). We will probably want a dedicated method for clarity and if the migrations workflow changes.

Reply:
Cool to see how the components can be used in a slightly different way to accomplish very different behavior 👍

Reply:
Oh, just noticed it shouldn't be called `sql_migration`, we're in the core.